externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala

```diff
@@ -84,34 +89,10 @@ object ResultSetUtil {
         case _: InterruptedException => fetched.getNextToken == null
       }
     } while (
-      fetched.getNextToken != null &&
+      System.currentTimeMillis() - fetchStart < fetchTimeout &&
```

Comment:
Will an exception be thrown if it times out, or will the state of the operation be the timeout state?

Reply:
Nope, then Flink engine would stop fetching more rows and return the result set.